### ISSUE TITLE
Impossible testing with pexpect when external printer used

### ISSUE THREAD

**Issue by godzie44** [kkawakam/rustyline#703]
I'm using rustyline and python pexpect for integration testing. This worked fine, until i'm try to use an external_printer. With external_printer pextect.send do nothing.

## Steps to reproduce

This can be reproduced on `external_print` example:

```python
>>> import pexpect
>>> proc = pexpect.spawn('./target/debug/examples/external_print')
>>> proc.sendline('123')
>>> print(proc.readline())
b'\x1b[?2004h\r\x1b[K> \r\x1b[2C\r\x1b[KExternal message #0\r\n'
>>> print(proc.readline())
b'\r\x1b[K> \r\x1b[2C\r\x1b[KExternal message #1\r\n'
>>> print(proc.readline())
```

and so on, expected output "Line: 123" never reads.

## Workaround

In my app i use cfg conditional checks for determine test environment or not, then in test env i replace `external_printer.print` to `println!`, but would be nice to have a better workaround...

**Comment by gwenn** [comment]
#690 reveals that `rustyline` doesn't behave correctly with `expect` / `pexpect`.
So maybe we should introduce a feature X (for testing) that deactivates input buffering ?
And maybe when this feature X is activated, external_printer are disabled ?

**Comment by godzie44** [comment]
Yep i think better if feature for testing will be on `rustyline` side.
But there is still a problem, that output observable by user and output observable by integration tests may be different.

**Comment by gwenn** [comment]
https://github.com/prompt-toolkit/python-prompt-toolkit/blob/0aa8a9762a51289e753a6e311edcb40b65224c66/src/prompt_toolkit/input/typeahead.py
> To support type ahead, this module will store all the key strokes that were
read too early, so that they can be feed into to the next `prompt()` call
